x-pack/test/security_solution_endpoint/services/endpoint.ts

```diff
@@ -291,4 +299,67 @@ export class EndpointTestResources extends FtrService {
   > {
     return installOrUpgradeEndpointFleetPackage(this.kbnClient);
   }
+
+  /**
+   * Fetch (GET) the details of an endpoint
+   * @param endpointAgentId
+   */
+  async fetchEndpointMetadata(endpointAgentId: string): Promise<HostInfo> {
+    const metadata = this.supertest
+      .get(HOST_METADATA_GET_ROUTE.replace('{id}', endpointAgentId))
+      .set('kbn-xsrf', 'true')
+      .send()
+      .expect(200)
+      .then((response) => response.body as HostInfo);
+
+    return metadata;
+  }
+
+  /**
+   * Sends an updated metadata document for a given endpoint to the datastream and waits for the
+   * update to show up on the Metadata API (after transform runs)
+   */
+  async sendEndpointMetadataUpdate(
+    endpointAgentId: string,
+    updates: DeepPartial<HostMetadata> = {}
+  ): Promise<HostInfo> {
+    const currentMetadata = await this.fetchEndpointMetadata(endpointAgentId);
+    const generatedMetadataDoc = new EndpointDocGenerator().generateHostMetadata();
+
+    const updatedMetadataDoc = merge(
+      { ...currentMetadata.metadata },
+      // Grab the updated `event` and timestamp from the generator data
+      {
+        event: generatedMetadataDoc.event,
+        '@timestamp': generatedMetadataDoc['@timestamp'],
+      },
+      updates
```

Comment:
`updates` is an instance of the `HostMetadata` type, which includes the `event` and `@timestamp` properties. Consider moving `updates` above the assignment of `event` and `@timestamp` if these properties should always have the values from the `generatedMetadataDoc`.

Reply:
`updates` is allowed to override the two properties above. By default, we set up the update, but the caller can override anything on it.